### ISSUE TITLE
A failed execution case

### ISSUE THREAD

**Issue by erxiaozhou** [wasmerio/wasmer#3007]
### Describe the bug

On my AMD CPU x86_64,
There is a failed execution when using wasmer to run the test case [test-case.zip](https://github.com/bytecodealliance/wasmtime/files/8980520/test-case.zip). 
The tested wasmer version is 2.3.0 or current .

```sh
wasmer run <test_case_name>
```

### Expected behavior

![image](https://user-images.githubusercontent.com/32102519/177274918-27b444a6-e14c-488d-8884-a3a9aa75a033.png)

### Actual behavior
![image](https://user-images.githubusercontent.com/32102519/177275103-47fe209a-2687-4c21-a4e0-c00be39eb56d.png)

**Comment by epilys** [comment]
Couldn't reproduce this on arch64/M1. Will try on amd64 next.

**Comment by erxiaozhou** [comment]
> Couldn't reproduce this on arch64/M1. Will try on amd64 next.

Yes, this is expected. it seems that the bug is x86 specific. This bug may have a similar root cause of [CVE-2022-31104](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-31104).

**Comment by ptitSeb** [comment]
Oh, so it's probably an issue with Cranelift, and the solution would be to update to v0.85.1
